reproject/utils.py

```diff
@@ -133,3 +136,103 @@ def parse_output_projection(output_projection, shape_out=None, output_array=None
         raise ValueError("The shape of the output image should not be an "
                          "empty tuple")
     return wcs_out, shape_out
+
+
+def _block(reproject_func, array_in, wcs_in, wcs_out_sub, shape_out, i_range, j_range,
+           return_footprint):
+    # i and j range must be passed through for multiprocessing to know where to reinsert patches
+    result = reproject_func(array_in, wcs_in, wcs_out_sub,
+                            shape_out=shape_out, return_footprint=return_footprint)
+
+    res_arr = None
+    res_fp = None
+
+    if return_footprint:
+        res_arr, res_fp = result
+    else:
+        res_arr = result
+
+    return {'i': i_range, 'j': j_range, 'res_arr': res_arr, 'res_fp': res_fp}
+
+
+def reproject_blocked(reproject_func, array_in, wcs_in, shape_out, wcs_out, block_size,
+                      output_array=None,
+                      return_footprint=True, output_footprint=None, parallel=True):
+    if output_array is None:
+        output_array = np.zeros(shape_out, dtype=float)
+    if output_footprint is None and return_footprint:
+        output_footprint = np.zeros(shape_out, dtype=float)
+
+    # setup variables needed for multiprocessing if required
+    proc_pool = None
+    blocks_futures = []
+
+    if parallel or type(parallel) is int:
```

Comment:
I think we want `and` here?  Otherwise `parallel=0` will try to work with 0 workers

Reply:
This might need tightening up, I was imagining parellel=True as being let reproject auto determine threading, but as you say zero would lead to issues.